obs-studio-client/source/controller.cpp

```diff
@@ -193,37 +172,103 @@ bool is_process_alive(ProcessInfo pinfo) {
 }
 
 bool kill(ProcessInfo pinfo, uint32_t code, uint32_t& exitcode) {
-	SetLastError(ERROR_SUCCESS);
-	bool suc = TerminateProcess(reinterpret_cast<HANDLE>(pinfo.handle), code);
-	DWORD errorCode = GetLastError();
-	return (errorCode == ERROR_SUCCESS);
+	return TerminateProcess(reinterpret_cast<HANDLE>(pinfo.handle), code);
 }
 
 std::string get_working_directory() {
-	DWORD dwRequiredSize = 0;
-	LPTSTR lpBuffer;
-
-	dwRequiredSize = GetCurrentDirectory(0, NULL);
-
-	lpBuffer = new TCHAR[dwRequiredSize];
+	DWORD dwRequiredSize = GetCurrentDirectory(0, NULL);
+	std::basic_string<TCHAR> lpBuffer(dwRequiredSize, TCHAR());
 
-	GetCurrentDirectory(dwRequiredSize, lpBuffer);
+	GetCurrentDirectory(dwRequiredSize, &lpBuffer[0]);
 
 #ifdef UNICODE
-	std::string result(from_utf16_wide_to_utf8(lpBuffer, dwRequiredSize));
+	std::string result(from_utf16_wide_to_utf8(lpBuffer.data()));
 
-	delete lpBuffer;
 	return result;
 #else
+	/* Remove the explicit null-terminator */
+	lpBuffer.resize(dwRequiredSize - 1);
 	return lpBuffer;
 #endif
 }
 
+std::string get_temp_directory() {
+	std::basic_string<TCHAR> tmp(MAX_PATH + 1, TCHAR());
+	GetTempPath(MAX_PATH + 1, &tmp[0]);
+
+#ifdef UNICODE
+	DWORD tmp_len = GetLongPathName(&tmp[0], NULL, 0);
+	tmp.resize(tmp_len);
+
+	/* Note that this isn't a hack to use the same buffer,
+	 * it's explicitly meant to be used this way per MSDN. */
+	GetLongPathName(&tmp[0], &tmp[0], tmp_len);
+
+	return from_utf16_wide_to_utf8(tmp.data());
+#else
+	/* Non-unicode path won't give us above MAX_PATH ever. */
+	/* Shrink and remove the explicit null-terminator */
+	tmp.resize(MAX_PATH);
```

Comment:
We should probably opt for a wide-char to utf-8 conversion here anyway and explicitly call GetLongPathNameW.